### PR TITLE
Add data validation and compression to CI

### PR DESCRIPTION
**Pull request by joanise**
Also
 - refactor CI so we have one long recipe, so that we only install stuff once to do all the tests, but still let every test run before failing (this PR replaces #66 for that, with a further improved solution)
 - bug fix: remove the timestamp from the .json.gz files
 - commit the .json.gz files into the repo

### CODE REVIEW

**Comment by dlothian** [comment]
Is this supposed to run with all pushes or just to main?

**Comment by joanise** [comment]
The test workflow should run on all pushes and PRs, because we want to run the test suites on branches and PRs to know whether they're safe to merge into main.

**Comment by dlothian** [comment]
> The test workflow should run on all pushes and PRs, because we want to run the test suites on branches and PRs to know whether they're safe to merge into main.

right that makes sense to me, I had assumed it would be applied for PRs into main and not with each push in every branch, since for example I do mostly UI stuff and push a lot of smaller commits. I don't think it actively affects anything for me though if it does push every time?

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -14,32 +14,61 @@ jobs:
...
         run: npx npm run-script e2e
       - uses: actions/upload-artifact@v3
-        if: always()
+        if: steps.e2e.outcome == 'success'
         with:
           name: playwright-report
           path: playwright-report/
           retention-days: 30
+
+      - name: Fail if any Validation Step failed
```
Comment (roedoejet) [inline .github/workflows/tests.yaml:66]: Nice! I really like this. I was wondering how you were going to cause the action to fail eventually

**Review by roedoejet — APPROVED** [review]
Awesome, I really like this @joanise !

**Comment by joanise** [comment]
> right that makes sense to me, I had assumed it would be applied for PRs into main and not with each push in every branch, since for example I do mostly UI stuff and push a lot of smaller commits. I don't think it actively affects anything for me though if it does push every time?

Actions only run when you push, so if you do a bunch of commits and then push, that's just one run of the action.

We can also configure it differently and run only on PR, but I really like the safety net of getting the output of the actions on every push.

**Comment by dlothian** [comment]
> > right that makes sense to me, I had assumed it would be applied for PRs into main and not with each push in every branch, since for example I do mostly UI stuff and push a lot of smaller commits. I don't think it actively affects anything for me though if it does push every time?
> 
> Actions only run when you push, so if you do a bunch of commits and then push, that's just one run of the action.
> 
> We can also configure it differently and run only on PR, but I really like the safety net of getting the output of the actions on every push.

One time in 2020 when I was working on my personal laptop and had a lot of commits but hadn't pushed them and then I updated my terrible computer and it wiped everything and I lost all of those commits and I have pushed a little bit like a manic since lol. But that's alright, I think the extra safety is totally fine, I'm developing locally so those tests running I don't think will effect me actively.

**Pull request merged**